Makefile.rules

```diff
@@ -990,10 +990,23 @@ testbugfix: all
           ReadGapRoot( "tst/testbugfix.g" );' | $(TESTGAP) | \
             tee `date -u +dev/log/testbugfix2_%Y-%m-%d-%H-%M` )
 
-testlibgap: libgap.la obj/tst/testlibgap/basic.lo 
-	$(QUIET_LINK)$(LINK) $(GAP_LDFLAGS) obj/tst/testlibgap/basic.lo libgap.la -o test-libgap
-	./test-libgap -A -l $(top_srcdir) -m 32m -q -T --nointeract > basic.out
-	diff $(top_srcdir)/tst/testlibgap/basic.expect basic.out
+LIBGAPTESTS := $(addprefix tst/testlibgap/,basic wsload wscreate)
+
+# run a test in tst/testlibgap
+tst/testlibgap/%: obj/tst/testlibgap/%.lo obj/tst/testlibgap/common.lo libgap.la
+	$(QUIET_LINK)$(LINK) $(GAP_LDFLAGS) $^ -o $@
+
+clean: clean-testlibgap
+clean-testlibgap:
+	rm -f ${LIBGAPTESTS}
+
+# run all the tests in tst/testlibgap
+testlibgap: ${LIBGAPTESTS}
+	$(foreach v,$^,$(v) -A -l $(top_srcdir) -m 32m -q -T --nointeract >$(v).out; diff $(top_srcdir)/$(v).expect $(v).out;)
```

Comment:
it's make's [foreach](https://www.gnu.org/software/make/manual/html_node/Foreach-Function.html). I can probably format this line better, split it, if you like...

Reply:
minimum workspace size has been set in @markuspf 's original line in `Makefile.rules` to run the only available libgap's test.

Reply:
I think reformatting this to cover multiple lines would improve the readability. I'd also drop the `-m 32m` -- if there was a reason for it, then it can be added back, but that time with a comment explaining it.

Other than that, this PR seems ready to me. Thank you for your persistence!

Reply:
Thanks, fixed  in the latest commit.